docs/typescript.md

```diff
@@ -1,29 +1,73 @@
 ---
 layout: default
-title: '[EXPERIMENTAL] Using TypeScript with DNSControl'
+title: Using TypeScript with DNSControl
 ---
 
-# [EXPERIMENTAL] Using TypeScript with DNSControl
+# Using TypeScript with DNSControl (experimental)
 
-> **NOTE**: This feature is currently experimental. Type information may change in future releases, and any release (even a patch release) could introduce new type-checking errors into your `dnsconfig.js` file. If you plan to use TypeScript, please file bug reports for any issues you encounter, and avoid manually running type-checking as part of your deployment process until the “experimental” label is removed.
+## What is this?
 
-While DNSControl does not support TypeScript syntax in `dnsconfig.js`, you can still use TypeScript’s features in editors which support it (including Visual Studio Code). To set up TypeScript support in Visual Studio Code, follow these steps:
-First, you’ll need to grab the `dnscontrol.d.ts` file corresponding to your version of DNSControl. You can manually download this file off of GitHub, or you can use the following command:
+Would you like your editor to support auto-completion and other advanced IDE
+features when editing dnsconfig.js?  Yes you can!
+
+While DNSControl does not support TypeScript syntax in `dnsconfig.js`, you can
+still use TypeScript’s features in editors which support it.
+
+If you’re using VS Code (or another editor that supports TypeScript), you
+should now be able to see the type information in your `dnsconfig.js` file as
+you type. Hover over record names to read their documentation without having to
+open the website!
+
+## How to activate
+
+To set up TypeScript support in Visual Studio Code, follow these steps:
+
+1. Run this command to generate the file `types-dnscontrol.d.ts`.
 
 ```bash
 dnscontrol write-types
 ```
 
-When run, `dnscontrol write-types` will create a `dnscontrol.d.ts` file in the current directory, overwriting an existing file if it exists. The file will use the type information corresponding to the current version of `dnscontrol`, so you can be confident that everything in the type declarations are consistent with DNSControl functionality. That does mean that you should re-run `dnscontrol write-types` when you update DNSControl, though!
+This file has all the information your editor or IDE needs.  It must be in the same directory as the `dnsconfig.js` file you are editing.
+
+NOTE: Re-run the `dnscontrol write-types` command any time you upgrade
+dnscontrol. Because it is generated from the command, it will always be correct
+for the version of DNSControl you are using.
```

Comment:
This is a little unclear in my eyes (but I’m not sure if my suggestion makes it any better). I wanted to clarify that it’s downloaded from the Internet, and that it matches the current version of DNSControl rather than just grabbing the latest version.

```suggestion
NOTE: Re-run the `dnscontrol write-types` command any time you upgrade
dnscontrol. The file will always match the version of DNSControl that is
currently running, regardless of what the latest version is.
```

Reply:
Oh! You are correct!  I forgot that the code downloads the latest version.  Would it be better if it downloaded the version that is appropriate for this version of dnscontrol?

There's an easy way to implement that.  Go can embed the file into the binary. The write-types command would write the contents to the file instead of downloading it.

Reply:
> Oh! You are correct! I forgot that the code downloads the latest version. Would it be better if it downloaded the version that is appropriate for this version of dnscontrol?

It should be downloading the version corresponding to the commit SHA used to build DNSControl. I didn’t embed it partially because I don’t really know how to do so but also because it’s around 100kB uncompressed and I wouldn’t want to bloat the binary too much (although I guess it isn’t too bad compared to the ~40-50MB total binary size)

Reply:
I'm ok with 100kB being added.  This way people that build and compile their own binaries will be supported.

Reply:
Will open a PR to do that then.